fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MultidimensionalIndexScanComparisons.java

```diff
@@ -0,0 +1,271 @@
+            } else {
+                detailsBuilder.add("suffix comparisons: {{dcomparisons" + d + "}}");
+                attributeMapBuilder.put("dcomparisons" + d, Attribute.gml(suffixScanComparisons.toString()));
+            }
+        }
+
+        tupleRange = suffixScanComparisons.toTupleRangeWithoutContext();
+        if (tupleRange != null) {
+            detailsBuilder.add("suffix: " + tupleRange.getLowEndpoint().toString(false) + "{{slow}}, {{shigh}}" + tupleRange.getHighEndpoint().toString(true));
+            attributeMapBuilder.put("slow", Attribute.gml(tupleRange.getLow() == null ? "-∞" : tupleRange.getLow().toString()));
+            attributeMapBuilder.put("shigh", Attribute.gml(tupleRange.getHigh() == null ? "∞" : tupleRange.getHigh().toString()));
+        } else {
+            detailsBuilder.add("suffix comparisons: {{scomparisons}}");
+            attributeMapBuilder.put("scomparisons", Attribute.gml(suffixScanComparisons.toString()));
+        }
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedTo() {
+        final ImmutableSet.Builder<CorrelationIdentifier> correlatedToBuilder = ImmutableSet.builder();
+        correlatedToBuilder.addAll(prefixScanComparisons.getCorrelatedTo());
+        correlatedToBuilder.addAll(dimensionsScanComparisons.stream()
+                .flatMap(dimensionScanComparison -> dimensionScanComparison.getCorrelatedTo().stream()).iterator());
+        correlatedToBuilder.addAll(suffixScanComparisons.getCorrelatedTo());
+        return correlatedToBuilder.build();
+    }
+
+    @Nonnull
+    @Override
+    public IndexScanParameters rebase(@Nonnull final AliasMap translationMap) {
+        return translateCorrelations(TranslationMap.rebaseWithAliasMap(translationMap));
+    }
+
+    @Override
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    public boolean semanticEquals(@Nullable final Object other, @Nonnull final AliasMap aliasMap) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        final MultidimensionalIndexScanComparisons that = (MultidimensionalIndexScanComparisons)other;
+
+        if (!prefixScanComparisons.semanticEquals(that.prefixScanComparisons, aliasMap)) {
+            return false;
+        }
+        if (dimensionsScanComparisons.size() != that.dimensionsScanComparisons.size()) {
+            return false;
+        }
+        for (int i = 0; i < dimensionsScanComparisons.size(); i++) {
+            final ScanComparisons dimensionScanComparison = dimensionsScanComparisons.get(i);
+            final ScanComparisons otherDimensionScanComparison = that.dimensionsScanComparisons.get(i);
+            if (!dimensionScanComparison.semanticEquals(otherDimensionScanComparison, aliasMap)) {
+                return false;
+            }
+        }
```

Comment:
Yes it does for this particular operation as the scan comparisons only really mean something if you also consider the definition of the underlying index. An even though an index `(a, {d1, d2, d3}, b)` is conceptually equivalent to `(a, {d3, d1, d2}, b)` in terms of how it can be queried, the underlying storage layout of these two indexes is inherently incompatible to each other.